### ISSUE TITLE
need an option to set ingress.class

### ISSUE THREAD

**Issue by ivukotic** [ssl-hep/ServiceX#232]
On most clusters this won't be nginx. So this is a "must have" option.

**Comment by AndrewEckart** [comment]
The `kubernetes.io/ingress.class` annotation [has been deprecated](https://kubernetes.io/docs/concepts/services-networking/ingress/#deprecated-annotation) as of K8s version 1.18 (it was never official).

Kubernetes 1.18 introduced a new API for specifying the class of an Ingress:
- There is a new resource called an IngressClass
- An IngressClass can be marked as the default by annotating it with `ingressclass.kubernetes.io/is-default-class: true`.
- When a new Ingress is created on the cluster, it will be given the default IngressClass unless it has an `ingressClassName` field specified. The value of `ingressClassName` must be the **name of an Ingress Class resource** on the cluster, which may not necessarily just be `nginx`.

Long story short, for newer clusters which have set up a default IngressClass, we can remove this annotation entirely - this is all handled automatically. For newer clusters without a default ingress class, we should support setting the `ingressClassName` field. We can also consider supporting older clusters via the deprecated annotation.

**Comment by ivukotic** [comment]
Please keep in mind our current cluster is 1.17.
Ilija

**Comment by AndrewEckart** [comment]
> Please keep in mind our current cluster is 1.17.

Lincoln said he plans to update it to the latest version (1.20) on Monday, although he's not planning on configuring an IngressClass.
